iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/chunk_distributor.inl

```diff
@@ -132,23 +132,67 @@ inline bool ChunkDistributor<ChunkDistributorDataType>::hasStoredQueues() const
 template <typename ChunkDistributorDataType>
 inline void ChunkDistributor<ChunkDistributorDataType>::deliverToAllStoredQueues(mepoo::SharedChunk chunk) noexcept
 {
-    typename MemberType_t::LockGuard_t lock(*getMembers());
+    typename ChunkDistributorDataType::QueueContainer_t remainingQueues;
+    {
+        typename MemberType_t::LockGuard_t lock(*getMembers());
+
+        // send to all the queues
+        for (auto& queue : getMembers()->m_queues)
+        {
+            bool isBlockingQueue =
+                (getMembers()->m_subscriberTooSlowPolicy == SubscriberTooSlowPolicy::WAIT_FOR_SUBSCRIBER
+                 && queue->m_queueFullPolicy == QueueFullPolicy::BLOCK_PUBLISHER);
 
-    // send to all the queues
-    for (auto& queue : getMembers()->m_queues)
+            if (!deliverToQueue(queue.get(), chunk) && isBlockingQueue)
+            {
+                remainingQueues.emplace_back(queue);
+            }
+        }
+    }
+
+    // busy waiting until every queue is served
+    while (!remainingQueues.empty())
     {
-        deliverToQueue(queue.get(), chunk);
+        std::this_thread::yield();
+        {
+            // create intersection of current queues and remainingQueues
+            typename MemberType_t::LockGuard_t lock(*getMembers());
+            typename ChunkDistributorDataType::QueueContainer_t queueIntersection(remainingQueues.size());
+            std::sort(getMembers()->m_queues.begin(), getMembers()->m_queues.end());
+            std::sort(remainingQueues.begin(), remainingQueues.end());
```

Comment:
Why do we need to sort here? `set_intersection` does that for you.

Reply:
According to: https://en.cppreference.com/w/cpp/algorithm/set_intersection the input ranges have to be sorted.This states the same: http://www.cplusplus.com/reference/algorithm/set_intersection/

To quote:
Constructs a sorted range beginning in the location pointed by result with the set intersection of the two sorted ranges [first1,last1) and [first2,last2).